### ISSUE TITLE
Dynamic source declaration

### ISSUE THREAD

**Issue by apobekiaris** [fsprojects/Paket#3750]
### Description

My source is in the form https://nuget.devexpress.com/MySecret/api. What the optimum way to use it with paket?

I tried just to edit both dependencies and lock files and replace it manually but this brings other problems e.g. with Devops NugetCache so maybe there is a better way and I miss it?

**Comment by forki** [comment]
Why do you need to touch the lock file? Shouldn't it just work in deps file?

**Comment by apobekiaris** [comment]
the lock file contains the  https://nuget.devexpress.com/MySecret/api i do not wish to spent all that paket-install time after i replace it in the dependencies. What I wish is that I can hook up an enviromental variable as my source.

**Comment by forki** [comment]
isn't the real issue that they put secrets in a URL?

**Comment by apobekiaris** [comment]
yes, but it is what it is, this is a commercial product feed which I have to use

**Comment by forki** [comment]
for reference: https://documentation.devexpress.com/GeneralInformation/116042/Installation/Install-DevExpress-Controls-Using-NuGet-Packages/Obtain-Your-NuGet-Feed-URL shows their system.

TBH: I think they should fix their system. This is not a paket issue.

that said. we already support placeholders for username and password (see https://fsprojects.github.io/Paket/nuget-dependencies.html#NuGet-feeds). we could probably accept PRs that implement `%PRIVATE_FEED_URL%`

**Comment by damianh** [comment]
This mechanism of accessing secured feeds is just plain broken and should be deprecated. However the horse truly bolted on this one. Sorry @forki , if you want packet to be used for people (stuck) in this situation, you'll have to compromise on the principals :( . Yeah that means `%PRIVATE_FEED_URL%`.

**Comment by apobekiaris** [comment]
DevExpress which I also work for, be sure, analyze all decisions it takes in many levels from different people. Having something perfect in theory as you suggested may not be optimum in all cases. In addition the security of each url can be analyzed for abuse from DevExpress side. So, said that, I really meant that DevExpress may not have an issue here and the decision is optimum given their customer base or other factors we are not aware.

So as you certain more flexible and you want people to use paket consider adding this small enhancement which however is common to thousand of people as you already discovered by your self.

Thnks

**Comment by forki** [comment]
They don't have an issue because they don't care for web standards. Authorization headers are there for a reason abd security tokens should really go there.

Again, we are still open for contributions. I already marked it up-for-grabs. 
And if it's too hard to do then I can help.

**Comment by forki** [comment]
Corresponding tweet: https://twitter.com/sforkmann/status/1203200907678224390?s=19
